### PR TITLE
Update to Dell System Update

### PR DESCRIPTION
**Pull request by djjudas21**
Update to Dell System Update, which implies dropping support for RHEL 5 and older (because they are no longer supported by the vendor).

Tested on various Dell hardware running RHEL 6 and RHEL 7.

I believe this supersedes #57 

This definitely supersedes #77 as this contains lint fixes and dropping the CentOS 5 test

### CODE REVIEW

**Comment by djjudas21** [comment]
@raphink Are you able to help with the tests for this? I don't understand why some are failing and some are OK, I don't know enough about rspec.

**Comment by djjudas21** [comment]
I think this is ready to merge now

**Comment by legooolas** [comment]
I'd like to see this merged too please, so that I can stop using my hacky branch which @djjudas21 has fixed up in this pull request  :)

**Comment by djjudas21** [comment]
Hello?

**Comment by raphink** [comment]
@djjudas21 sorry for the delay, I'll have a look.

File: manifests/hwtools/package.pp [hunk 1]
```diff
@@ -7,8 +8,29 @@
     }
 
     'RedHat': {
-      package{['libsmbios', 'smbios-utils-bin']:
-        ensure => latest,
+      #package{['libsmbios', 'smbios-utils-bin']:
+      #  ensure => latest,
+      #}
```
[the remark this answers is not in the document]
Comment (djjudas21) [reply manifests/hwtools/package.pp]: Fixed in 8223f47

File: manifests/openmanage.pp [hunk 2]
```diff
@@ -56,6 +61,17 @@
...
+      # on OM 8.3, for details see
+      # https://www.mail-archive.com/search?l=linux-poweredge@dell.com&q=subject:%22\[Linux\-PowerEdge\]+srvadmin+8.3.0+problems+with+dsm_sa_datamgrd%22&o=newest&f=1
+      file_line{'comment-vil7':
+        path    => '/opt/dell/srvadmin/etc/srvadmin-storage/stsvc.ini',
+        line    => '; vil7=dsm_sm_psrvil',
+        match   => 'vil7=dsm_sm_psrvil',
+        before  => Service['dataeng'],
+        notify  => Service['dataeng'],
+        require => Package['srvadmin-storageservices'],
+      }
```
[the remark this answers is not in the document]
Comment (djjudas21) [reply manifests/openmanage.pp]: This piece of code is one I grabbed from another contributor. If modelled using an `ini_setting` resource, does this mean I just need to do

```puppet
ini_setting { 'vil7':
  ensure => absent,
}
```

[the remark this answers is not in the document]
Comment (raphink) [reply manifests/openmanage.pp]: With the target and potentially the section, yes

[the remark this answers is not in the document]
Comment (djjudas21) [reply manifests/openmanage.pp]: Fixed in d309047

**Comment by djjudas21** [comment]
Hmm, this is failing but I don't know why as I'm not familiar with Travis

**Comment by djjudas21** [comment]
CI test failing only only RedHat with

```
  1) dell::openmanage on redhat-6-x86_64 should compile into a catalogue without dependency cycles
     Failure/Error: it { should compile.with_all_deps }
       error during compilation: Evaluation Error: Error while evaluating a Resource Statement, Unknown resource type: 'ini_setting' at /home/travis/build/camptocamp/puppet-dell/spec/fixtures/modules/dell/manifests/openmanage.pp:67:7 on node travis-job-camptocamp-puppet-dell-316417344.travisci.net
     # ./spec/classes/dell__openmanage_spec.rb:10:in `block (4 levels) in <top (required)>'
```

Bit the `puppetlabs/inifile` is specified everywhere that `puppetlabs/stdlib` so I don't know what the problem is. Could you take a look please @raphink - my rspec-fu is not strong - thanks :+1:

**Comment by raphink** [comment]
You need to the module to `.fixtures.yml` as well for unit tests

**Comment by djjudas21** [comment]
Oh, I didn't even know that file existed. Added now... let's see how the tests look

**Comment by raphink** [comment]
Almost there it seems. The parameter is `path`, not `target` (my bad for giving you that erroneous info).

**Comment by djjudas21** [comment]
And my bad for not RTFM :wink:

**Comment by djjudas21** [comment]
Awesome, thanks. May I ask you to tag a new release, please?

**Comment by raphink** [comment]
There, releasing 1.0.0 as it breaks compatibility (several OSes not supported anymore)

**Pull request merged**